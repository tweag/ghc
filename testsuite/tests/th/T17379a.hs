{-# LANGUAGE TemplateHaskell #-}
{-# LANGUAGE GADTSyntax #-}

module T17379a where

import Language.Haskell.TH

$(let typ = mkName "T" in pure [ DataD [] typ [] Nothing [GadtC [] [] (ConT typ)] [] ])
