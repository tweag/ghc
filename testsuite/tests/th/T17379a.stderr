
T17379a.hs:8:2: error:
    GadtC must have at least one constructor name
    When splicing a TH declaration: data T where :: T
